Spoony-iOS/Spoony-iOS/Resource/Helper/PopupModifier.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  PopupModifier.swift
+//  Spoony-iOS
+//
+//  Created by 최주리 on 1/12/25.
+//
+
+import SwiftUI
+
+struct PopupModifier: ViewModifier {
+    let popup: PopupType
+    @Binding var isPresented: Bool
+    
+    func body(content: Content) -> some View {
+        ZStack {
+            content
+            if isPresented {
+                Color.black.opacity(0.6)
+                    .ignoresSafeArea()
+                    
+                PopupView(popup: popup, isPresented: $isPresented)
+            }
+        }
+    }
+}
+
+enum PopupType {
+    case useSpoon(action: () -> Void)
+    case reportSuccess(action: () -> Void)
+    case registerSuccess(action: () -> Void)
+}
+
+struct PopupView: View {
+    let popup: PopupType
+    @Binding var isPresented: Bool
+    
+    var body: some View {
+        VStack(spacing: 20) {
+            Spacer()
+                .frame(height: 20)
+            if let image {
+                Image(image)
+                    .resizable()
+                    .scaledToFit()
+                    .frame(width: 150, height: 150)
+            }
+            
+            Text(description)
+                .multilineTextAlignment(.center)
+                .font(.body1b)
+                .padding(.bottom, descriptionYOffset)
+            
+            HStack(spacing: 12) {
+                if let title = grayButtonTitle {
+                    SpoonyButton(
+                        style: .teritary,
+                        size: .xsmall,
+                        title: title,
+                        disabled: .constant(false)
+                    ) {
+                        isPresented = false
+                    }
+                    
+                    SpoonyButton(
+                        style: .secondary,
+                        size: .xsmall,
+                        title: blackButtonTitle,
+                        disabled: .constant(false)
+                    ) {
+                        confirmAction()
+                        isPresented = false
+                    }
+                } else {
+                    SpoonyButton(
+                        style: .secondary,
+                        size: .large,
+                        title: blackButtonTitle,
+                        disabled: .constant(false)
+                    ) {
+                        confirmAction()
+                        isPresented = false
+                    }
+                }
+            }
+        }
```

Comment:
지금 grayButton의 값이 있을 때와, 없을 때의 레이아웃을 분기 처리하고 있는 부분이 맞지요~? (버튼 2개와 1개의 분기)
지금 보면 있을 때와 없을 때는 `secondary` 버튼의 `size`에 차이가 있는 거 같아요

따라서 아래처럼 중복 코드를 개선하면 어떨까요?
```suggestion
                if let title = grayButtonTitle {
                    SpoonyButton(
                        style: .teritary,
                        size: .xsmall,
                        title: title,
                        disabled: .constant(false)
                    ) {
                        isPresented = false
                    }
                }
                SpoonyButton(
                    style: .secondary,
                    size: grayButtonTitle == nil ? .large : .xsmall,
                    title: blackButtonTitle,
                    disabled: .constant(false)
                ) {
                    confirmAction()
                    isPresented = false
                }
            }
        }
```